test/hotspot/jtreg/serviceability/attach/LongArgTest.java

```diff
@@ -0,0 +1,161 @@
+import java.io.InputStream;
+import java.io.IOException;
+
+import com.sun.tools.attach.VirtualMachine;
+import sun.tools.attach.HotSpotVirtualMachine;
+
+import jdk.test.lib.apps.LingeredApp;
+
+public class LongArgTest {
+
+    public static void main(String[] args) throws Exception {
+        LingeredApp app = null;
+        try {
+            app = LingeredApp.startApp();
+
+            test(app).run();
+
+            test(app)
+                .withLongValue()
+                .run();
+
+            test(app)
+                .withSuperLongValue()
+                .run();
+
+        } finally {
+            LingeredApp.stopApp(app);
+        }
+
+    }
+
+    // For simplicity, the test uses internal HotSpotVirtualMachine,
+    // sets/gets "HeapDumpPath" flag value (string flag, not validated by JVM).
+    private static class Test {
+        private LingeredApp app;
+        private String flagName = "HeapDumpPath";
+        private String flagValue = generateValue(5);
+
+        Test(LingeredApp app) {
+            this.app = app;
+        }
+
+        // Value length exceeds 1K.
+        Test withLongValue() {
+            flagValue = generateValue(1024 + 1);
+            return this;
+        }
+
+        // Value length exceeds 3K (1K * 3 args).
+        Test withSuperLongValue() {
+            flagValue = generateValue(3 * 1024 + 1);
+            return this;
+        }
+
+        void run() throws Exception {
+            System.out.println("======== Start ========");
+
+            HotSpotVirtualMachine vm = (HotSpotVirtualMachine)VirtualMachine.attach(String.valueOf(app.getPid()));
+
+            if (setFlag(vm)) {
+                String actualValue = getFlag(vm);
+
+                if (!flagValue.equals(actualValue)) {
+                    String msg = "Actual values is different: "
+                                + (actualValue == null
+                                    ? "null"
+                                    : ("size is " + actualValue.length() + " instead of " + flagValue.length()));
+                    System.out.println(msg);
+                    vm.detach();
+                    throw new RuntimeException(msg);
+                } else {
+                    System.out.println("Actual value matches: " + actualValue);
+                }
+            }
+
+            vm.detach();
+
+            System.out.println("======== End ========");
+            System.out.println();
+        }
+
+        // Sets the flag value, return true on success.
+        private boolean setFlag(HotSpotVirtualMachine vm) throws Exception {
+            BufferedReader replyReader = null;
+            try {
+                replyReader = new BufferedReader(new InputStreamReader(
+                    vm.setFlag(flagName, flagValue)));
+            } catch (IOException ex) {
+                System.out.println("OK: setFlag() thrown exception:");
+                ex.printStackTrace(System.out);
+                return false;
+            }
```

Comment:
Q: Is it always the case there is no need to close the `BufferedReader` in this context? Just want to understand.

Reply:
We can get exception here only if vm.setFlag() throws it. In the case BufferedReader is not yet constructed.